proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,797 @@
+...
+
+   [net5.0]:
+   Top-level Package      Requested             Resolved
+   > NuGet.Commands       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib         1.1.2                 1.1.2
+
+   Transitive Package                                                                   Resolved
+   > Microsoft.CSharp                                                                   4.0.1
+   > Microsoft.NETCore.Platforms                                                        1.1.0
+   > Microsoft.NETCore.Targets                                                          1.1.0
+...
+
+```
+
+### dotnet list package --include-transitive --json --flatten
+
+```json
+{
+  "version": 1,
+  "topLevelPackages": [
+       {
+            "id": "Microsoft.Extensions.Primitives",
+            "requestedVersion": "[1.0.0, 5.0.0]",
+            "resolvedVersion": "1.0.0"
+        },
+        {
+            "id": "NuGet.Commands",
+            "requestedVersion": "4.8.0-preview3.5278",
+            "resolvedVersion": "4.8.0-preview3.5278"
+        },
+        {
+            "id": "Text2Xml.Lib",
+            "requestedVersion": "[1.1.2, 2.0.0)",
+            "resolvedVersion": "1.1.2"
+        }
+    ],
+    "transitivePackages": [
+        {
+            "id": "Microsoft.CSharp",
+            "resolvedVersion": "4.0.1"
+        },
+        {
+            "id": "Microsoft.NETCore.Platforms",
+            "resolvedVersion": "1.1.0"
+        },
+        {
+            "id": "Microsoft.NETCore.Targets",
+            "resolvedVersion": "1.1.0"
+        }
+...
+    ]
+}
+```
+
+### Compatibility
+
+ We start with `version 1`, as long as we don't remove or rename then it'll be backward compatible. In case [we change version](https://stackoverflow.com/a/13945074) just add new properties, keep old ones even it's not used.
+
+### Out-of-scope
+
+* To avoid disk I/O, we won't support saving the machine-readable output to disk as part of this spec. The work-around is for the consumer to read from the console's stdout stream.
+* At this point, no other CLI commands (e.g. dotnet list reference) will be within scope for this feature.
+"--parsable" option needs separate spec.
+* Currently license info is not emitted from any cli command, it could be quite useful, we should consider in the future.
+
+## Prior Art
+
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevent to this proposal? -->
+
+* https://github.com/NuGet/Home/blob/dotnet-audit/proposed/2021/DotNetAudit.md#dotnet-audit---json
+
+* https://github.com/NuGet/Home/wiki/%5BSpec%5D-Machine-readable-output-for-dotnet-list-package
+
+## Feedback
+
+File NuGet.Client bugs in the [NuGet/Home](https://github.com/nuget/home/issues)
```

Comment:
I don't understand the benefit of this feedback section. When the design is under review, shouldn't feedback be in this pull request?

After it's merged, why file a bug? Depending on the feedback, it might be a feature or DCR. In fact, arguably any feedback will be a feature or DCR since whatever issue the customer has with the command once implemented will be working as designed, so design feedback after merge by definition can't be a bug?

Are we also trying to signal that pull requests that modify this document will be rejected without consideration?

Considering that Home/issues is the place for all product feedback, I personally don't see the benefit of calling it out in this design spec unless we're signaling that we want a non-normal process for feedback.